src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -325,6 +346,49 @@ public Task<IAsyncCursor<RawBsonDocument>> ResumeAsync(IReadBinding binding, Can
             return aggregateOperation.ExecuteAsync(binding, cancellationToken);
         }
 
+        /// <inheritdoc />
+        public void ReviewResumeParams()
+        {
+            // Docs: if the ChangeStream has a cached postBatchResumeToken and has returned all documents in the most recent batch(or the most recent batch was empty):
+            // The driver MUST set resumeAfter to the cached postBatchResumeToken.
+            // The driver MUST NOT set startAfter. If startAfter was in the original aggregation command, the driver MUST remove it.
+            // The driver MUST NOT set startAtOperationTime. If startAtOperationTime was in the original aggregation command, the driver MUST remove it.
+            // todo: should be implemented in scope of the CSHARP-2514.
+
+            // Docs: if the ChangeStream has a cached documentResumeToken:
+            // The driver MUST set resumeAfter to the cached documentResumeToken.
+            // The driver MUST NOT set startAfter. If startAfter was in the original aggregation command, the driver MUST remove it.
+            // The driver MUST NOT set startAtOperationTime. If startAtOperationTime was in the original aggregation command, the driver MUST remove it.
+            // todo:
+
+            // Docs: if the ChangeStream was created with startAfter:
+            // The driver MUST set resumeAfter to the value of the originally used startAfter.
+            // The driver MUST NOT set startAfter. The driver MUST remove it from the original aggregation command.
+            // The driver MUST NOT set startAtOperationTime.
+            if (_startAfter != null)
+            {
+                _resumeAfter = _startAfter;
+                _startAfter = null;
+                _startAtOperationTime = null;
+            }
+            // Docs: if the ChangeStream was created with resumeAfter:
```

Comment:
this question has been asked in the SPEC ticket: https://jira.mongodb.org/browse/SPEC-1227 partially.
We can have `_resumeAfter` value here which can be originally set in the `_startAfter` field. 
So, I'm not sure that below condition full satisfies to this: `if the ChangeStream was created with `resumeAfter`:`

Reply:
It's hard to talk about "was created with" if we are mutating the object.

If we had an object we don't mutate the the value of the field IS the original value.